lib/src/kbasesearchengine/parse/ObjectParser.java

```diff
@@ -57,21 +72,47 @@
         return guidToJson;
     }
 
-    public static GUID prepareGUID(ObjectTypeParsingRules parsingRules,
-            GUID guid, ObjectJsonPath path,
+    /** Appends the sub-object path and id to the specified GUID if the parsing
+     * rules define the extraction of data from a sub-object.
+     *
+     *
+     * @param parsingRules parsing rules
+     * @param guid GUID without the sub-object path and id
+     * @param path
+     * @param idConsumer
+     * @return GUID with sub-object path and id if data needs to be extracted
+     * from sub-object, lese GUID with just main object information.
+     */
+    public static GUID prepareGUID(
```

Comment:
Out of scope, but I really dislike that this method is public. We should (later) try to make it private and remove the id-defaulting-to-path stuff.

Reply:
Actually I think that's true about virtually every method in this class. I'd prefer to have one API call that takes the source data (+ whatever else is needed) and returns a single value object containing the resulting extracted data. I think all this stuff is batched anyway...

Reply:
I totally agree. If we reused an external package like JsonSurfer, we would end up with a single API call. I will leave this as-is for now and will make a note to come back to this post contractor meeting and re-evaluate the option to reuse external libs here and simplify the code that way.